src/mon/PGMap.cc

```diff
@@ -69,12 +69,15 @@ void PGMapDigest::encode(bufferlist& bl, uint64_t features) const
   encode(avail_space_by_rule, bl);
   encode(purged_snaps, bl);
   encode(osd_sum_by_class, bl, features);
+  if (v >= 5) {
```

Comment:
You don't need to check for v in encode, the encode is always the current version of the mon, so it must be 5 in that case, if it's earlier, you won't have the code on.